contracts/interchain-token-service/src/msg.rs

```diff
@@ -38,6 +39,17 @@ pub enum ExecuteMsg {
     #[permission(Governance)]
     RegisterChains { chains: Vec<ChainConfig> },
 
+    // Increase or decrease the supply for a given token and chain.
+    // If the supply is untracked, this command will attempt to set it.
+    // Errors if the token is not deployed to the specified chain, or if
+    // the supply modification overflows or underflows
+    #[permission(Elevated)]
+    ModifySupply {
+        chain: ChainNameRaw,
+        token_id: TokenId,
+        supply_modifier: SupplyModifier,
+    },
+
```

Comment:
I'd prefer having two entrypoints `IncreaseTokenSupply` and `DecreaseTokenSupply` for simplicity. The implementation could use a modify/update supply internally for reuse

Reply:
What would be the purpose of that? It will be more code and more complexity in the contract if we have two endpoints, so without a good reason to have two, I prefer one endpoint. A user isn't any more likely to mess something up as it is now versus if there are two endpoints.

Reply:
I don't see any benefit in splitting the endpoint either. Could you elaborate if you feel strongly about it @milapsheth?

Reply:
Not a dealbreaker, but I find it more readable to have two simpler entrypoints for other teams looking at it, and I don't like exposing the complexity to the user due to internal implementation reuse details.
```
IncreaseSupply {
        chain: ChainNameRaw,
        token_id: TokenId,
        amount: nonempty::Uint256,
    },
```
and the JSON looks like
`{"increase_supply": { "chain": "sui", "token_id": "...", "amount": "123" } }`

instead of
`{"modify_supply": { "chain": "sui", "token_id": "...", "supply_modifier": { "increase_supply": "123" } } }`

It's common to see this in other [APIs](https://docs.openzeppelin.com/contracts/2.x/api/token/erc20#ERC20-increaseAllowance-address-uint256-) too, granted it's not as bad as using a bool for the modifier

Reply:
hmm you make a good point. However, the users of this are just the operators of the contract, which is just our devops team, and the commands will likely be wrapped in a script. And this contract is already overly complex imo, so I prefer to err on the side of keeping the code simple here. If this API was more user facing, i.e. called by many third parties, external to our org, I would feel differently. Keeping it as is for now.